options.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+add_action( 'admin_init', 'amp_register_settings' );
+add_action( 'admin_menu', 'amp_custom_admin_menu' );
+
+/**
+ * Register the settings
+ */
+function amp_register_settings() {
+     register_setting(
+          'amp_options',  // settings section
+          'amp_canonical' // setting name
```

Comment:
We should pass in a third arg here, which at a minimum, specifies the `type`, `sanitize_callback`, and `default`. (see https://developer.wordpress.org/reference/functions/register_setting/#parameters)